### PR TITLE
Initial cut of facebook integration

### PR DESCRIPTION
**Pull request by bokelley**
First version based on initial Facebook OpenRTB spec. Parallelizes requests per FB recommendation.

### CODE REVIEW

File: adapters/facebook.go [hunk 1]
```diff
@@ -30,106 +32,147 @@ func (a *FacebookAdapter) FamilyName() string {
...
-		httpReq.Header.Add("Accept", "application/json")
+	result.statusCode = anResp.StatusCode
 
-		anResp, err := ctxhttp.Do(ctx, a.http.Client, httpReq)
-		if err != nil {
-			return nil, err
-		}
+	if anResp.StatusCode != 200 {
+		defer anResp.Body.Close()
+		body, _ := ioutil.ReadAll(anResp.Body)
```
Comment (asweeney86) [inline adapters/facebook.go]: This can be moved out since its used below as well

File: adapters/index.go [hunk 2]
```diff
@@ -137,7 +138,7 @@ func (a *IndexAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pb
 
 func NewIndexAdapter(config *HTTPAdapterConfig, externalURL string) *IndexAdapter {
 	a := NewHTTPAdapter(config)
-	redirect_uri := fmt.Sprintf("%s/setuid?bidder=indexExchange&uid=__UID__", externalURL)
+	redirect_uri := fmt.Sprintf("%s/setuid?bidder=indexExchange&uid=$UID", externalURL)
```
Comment (asweeney86) [inline adapters/index.go]: This seems like an independent change.  Was it picked up by mistake?

File: adapters/openrtb_util.go [hunk 3]
```diff
@@ -30,9 +30,13 @@ func makeOpenRTBGeneric(req *pbs.PBSRequest, bidder *pbs.PBSBidder, bidderFamily
 		}
 	}
 
+	cur := make([]string, 1)
+	cur[0] = "USD"
```
Comment (asweeney86) [inline adapters/openrtb_util.go]: Why do we need to hard code USD?

File: adapters/index.go (hunk 2 above)
Comment (bokelley) [inline adapters/index.go]: good catch. will rollback.

File: adapters/openrtb_util.go (hunk 3 above)
Comment (bokelley) [inline adapters/openrtb_util.go]: Facebook asks for it in their request, figured it was universal, but you're right. Will remove.

**Pull request merged**